include/SDL_mixer.h

```diff
@@ -22,12 +22,12 @@
 #ifndef SDL_MIXER_H_
 #define SDL_MIXER_H_
 
-#include "SDL_stdinc.h"
-#include "SDL_rwops.h"
-#include "SDL_audio.h"
-#include "SDL_endian.h"
-#include "SDL_version.h"
-#include "begin_code.h"
+#include "SDL2/SDL_stdinc.h"
```

Comment:
The include path changes are incorrect.

Reply:
I had errors with header not found, it now works for me and on github CI

Reply:
And your versions will fail with autotools and other build systems.
Those changes are wrong and can't be accepted. The SDL2 discovery
in cmake should be done properly, instead.

Reply:
> And your versions will fail with autotools and other build systems. Those changes are wrong and can't be accepted. The SDL2 discovery in cmake should be done properly, instead.

I will change that, do you have the links to these CI ? I would like to know how this is configured, to test in docker containers

Reply:
The following change to `CMakeLists.txt` works just fine for me:
```diff
diff --git a/CMakeLists.txt b/CMakeLists.txt
index 283f588..0934778 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 3.0.0)
 project(SDL_mixer C)
 
 # FIXME: make it able build against system codec libraries, too.
@@ -14,6 +14,9 @@ option(SUPPORT_MP3_MPG123 "Support loading MP3 music via MPG123" OFF)
 option(SUPPORT_MOD_MODPLUG "Support loading MOD music via modplug" OFF)
 option(SUPPORT_MID_TIMIDITY "Support TiMidity" OFF)
 
+find_package(SDL2 REQUIRED)
+include_directories(${SDL2_INCLUDE_DIRS} ${SDL2_INCLUDE_DIR})
+
 include_directories(include src src/codecs)
 
 add_library(SDL2_mixer SHARED)
@@ -64,5 +67,5 @@ if (SUPPORT_MID_TIMIDITY)
 endif()
 
 target_include_directories(SDL2_mixer PUBLIC include)
-target_link_libraries(SDL2_mixer PRIVATE SDL2)
+target_link_libraries(SDL2_mixer PRIVATE ${SDL2_LIBRARIES})
 
```

Reply:
> do you have the links to these CI ?

If the question is how to setup CI to use autotools (configure) or MS Visual Studio or Xcode,
I don't know.

The above patch to CMakeLists.txt is the correct way for SDL2 discovery, include directives
and linkage.

Reply:
> The following change to `CMakeLists.txt` works just fine for me:
> 
> ```diff
> diff --git a/CMakeLists.txt b/CMakeLists.txt
> index 283f588..0934778 100644
> --- a/CMakeLists.txt
> +++ b/CMakeLists.txt
> @@ -1,4 +1,4 @@
> -cmake_minimum_required(VERSION 2.8.12)
> +cmake_minimum_required(VERSION 3.0.0)
>  project(SDL_mixer C)
>  
>  # FIXME: make it able build against system codec libraries, too.
> @@ -14,6 +14,9 @@ option(SUPPORT_MP3_MPG123 "Support loading MP3 music via MPG123" OFF)
>  option(SUPPORT_MOD_MODPLUG "Support loading MOD music via modplug" OFF)
>  option(SUPPORT_MID_TIMIDITY "Support TiMidity" OFF)
>  
> +find_package(SDL2 REQUIRED)
> +include_directories(${SDL2_INCLUDE_DIRS} ${SDL2_INCLUDE_DIR})
> +
>  include_directories(include src src/codecs)
>  
>  add_library(SDL2_mixer SHARED)
> @@ -64,5 +67,5 @@ if (SUPPORT_MID_TIMIDITY)
>  endif()
>  
>  target_include_directories(SDL2_mixer PUBLIC include)
> -target_link_libraries(SDL2_mixer PRIVATE SDL2)
> +target_link_libraries(SDL2_mixer PRIVATE ${SDL2_LIBRARIES} ${SDL2_LIBRARY})
>  
> ```

It works for me too, and on CI: https://github.com/bensuperpc/SDL_mixer/actions/runs/1455167527

I have a warning that appeared (**but not on CI**):

```
cmake -S . -B build -G Ninja -DFIND_LOCAL_SDL2=OFF && ninja -C build
-- The C compiler identification is GNU 11.1.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    FIND_LOCAL_SDL2


-- Build files have been written to: SDL_mixer/build
ninja: Entering directory `build'
[21/21] Linking C static library libSDL2_mixer.a
```